corehq/apps/userreports/reports/builder/forms.py

```diff
@@ -438,40 +527,62 @@ def _make_report_filter(conf):
             }
 
         filter_configs = self.cleaned_data['filters']
-        return [_make_report_filter(f) for f in filter_configs]
+        filters = [_make_report_filter(f) for f in filter_configs]
+        if self.source_type == 'case':
+            self._convert_closed_filter_to_choice_list(filters)
+        return filters
+
+    @classmethod
+    def _convert_closed_filter_to_choice_list(cls, filters):
+        for f in filters:
+            if f['field'] == get_column_name('closed') and f['type'] == 'dynamic_choice_list':
+                f['type'] = 'choice_list'
+                f['choices'] = [
+                    {'value': 'True'},
+                    {'value': 'False'}
+                ]
 
     @property
     def _report_charts(self):
         return []
 
 
-class ConfigureNewBarChartReport(ConfigureNewReportBase):
-    group_by = forms.ChoiceField()
-    form_title = "Configure Bar Chart Report"
+class ConfigureBarChartReportForm(ConfigureNewReportBase):
+    group_by = forms.ChoiceField(label="Property")
+    report_type = 'chart'
 
-    def __init__(self, app_id, source_type, report_source_id, *args, **kwargs):
-        super(ConfigureNewBarChartReport, self).__init__(app_id, source_type, report_source_id, *args, **kwargs)
+    def __init__(self, report_name, app_id, source_type, report_source_id, existing_report=None, *args, **kwargs):
+        super(ConfigureBarChartReportForm, self).__init__(report_name, app_id, source_type, report_source_id, existing_report, *args, **kwargs)
         self.fields['group_by'].choices = self._group_by_choices
 
+        # Set initial value of group_by
+        if self.existing_report:
+            existing_agg_cols = existing_report.aggregation_columns
+            assert len(existing_agg_cols) < 2
+            if existing_agg_cols:
+                self.fields['group_by'].initial = self._get_property_from_column(existing_agg_cols[0])
+
     @property
-    def top_fieldset(self):
+    def container_fieldset(self):
         return crispy.Fieldset(
-            _(self.form_title),
-            'report_name',
+            _('Categories'),
             'group_by',
-            self.configuration_tables
+            self.filter_fieldset
         )
 
     @property
+    def aggregation_field(self):
+        return self.cleaned_data["group_by"]
```

Comment:
It might not be immediately obvious why this extra layer of encapsulation is being added; `ConfigureTableReportForm` overrides this functionality.